### ISSUE TITLE
Parser - Handle error nodes better

### ISSUE THREAD

**Issue by PrashanthCorp** [Microsoft/vscode-cosmosdb#471]
[Currently](https://github.com/Microsoft/vscode-cosmosdb/blob/master/src/mongo/grammar/visitors.ts#L67), we mainly print an error message with the location of the encountered error. 

A richer experience could have text highlighting, or something _at the position of erroneous text_. 

@fiveisprime  thoughts?

**Comment by StephenWeatherford** [comment]
@PrashanthCorp I think the next step is to select the error region in the text document, and see how well that works.  That might be good enough.

**Comment by StephenWeatherford** [comment]
Yes, we need to stop throwing in the visit, and instead implement something that can return the errors.

**Comment by StephenWeatherford** [comment]
Red squiggles would be ideal.

**Comment by StephenWeatherford** [comment]
I'm taking this one, got tired of current situation.
#652
